src/rpp/rpp/operators/retry_when.hpp

```diff
@@ -0,0 +1,158 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/observables/fwd.hpp>
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/strategy.hpp>
+#include <rpp/utils/constraints.hpp>
+#include <rpp/utils/utils.hpp>
+
+namespace rpp::operators::details
+{
+    template<rpp::constraint::observer     TObserver,
+             rpp::constraint::observable   TObservable,
+             rpp::constraint::decayed_type Notifier>
+    struct retry_when_impl_inner_strategy
+    {
+        using preferred_disposable_strategy = rpp::details::observers::none_disposable_strategy;
+
+        RPP_NO_UNIQUE_ADDRESS mutable TObserver observer;
+        RPP_NO_UNIQUE_ADDRESS TObservable       observable;
+        RPP_NO_UNIQUE_ADDRESS Notifier          notifier;
+
+        template<typename T>
+        void on_next(T&&) const
+        {
+            observable.subscribe(std::move(observer));
+        }
+
+        void on_error(const std::exception_ptr& err) const
+        {
+            observer.on_error(err);
+        }
+
+        void on_completed() const
+        {
+            observer.on_completed();
```

Comment:
observer can be moved out in this place (for example, in case of just(1) it would be on_next + on_completed)

Reply:
https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/rpp/rpp/operators/take_until.hpp#L62